tools/restatectl/src/build_info.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2024 -  Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Build information
+#![allow(dead_code)]
+
+use std::sync::OnceLock;
+
+/// The version of restate CLI.
+pub const RESTATE_CLI_VERSION: &str = env!("CARGO_PKG_VERSION");
```

Comment:
We might use a different name to differentiate from the CLI version variables. Maybe `RESTATE_CTL_VERSION`?